corehq/motech/dhis2/entities_helpers.py

```diff
@@ -317,27 +335,38 @@ def create_relationships(
                 'Unable to create DHIS2 relationship: The case {case} is not '
                 'registered in DHIS2.'
             ).format(case=supercase_trigger_info))
-
-        if relationship_config.supercase_to_subcase_dhis2_id:
-            relationship_spec = RelationshipSpec(
+        supercase_to_subcase_relationship_spec = RelationshipSpec(
                 relationship_type_id=relationship_config.supercase_to_subcase_dhis2_id,
                 from_tracked_entity_instance_id=supercase_tracked_entity_instance_id,
                 to_tracked_entity_instance_id=subcase_tracked_entity_instance_id
             )
-            create_relationship_at_dhis2(
-                requests=requests,
-                relationship_spec=relationship_spec
-            )
-        if relationship_config.subcase_to_supercase_dhis2_id:
-            relationship_spec = RelationshipSpec(
+        subcase_to_supercase_relationship_spec = RelationshipSpec(
                 relationship_type_id=relationship_config.subcase_to_supercase_dhis2_id,
                 from_tracked_entity_instance_id=subcase_tracked_entity_instance_id,
                 to_tracked_entity_instance_id=supercase_tracked_entity_instance_id,
             )
-            create_relationship_at_dhis2(
-                requests=requests,
-                relationship_spec=relationship_spec
-            )
+        ensure_relationship_exists(
+            requests=requests,
+            relationship_spec=supercase_to_subcase_relationship_spec,
+            existing_relationships=tracked_entity_relationships
+        )
+        ensure_relationship_exists(
+            requests=requests,
+            relationship_spec=subcase_to_supercase_relationship_spec,
+            existing_relationships=tracked_entity_relationships
+        )
+
+def ensure_relationship_exists(requests, relationship_spec, existing_relationships):
+    """Checks to see if `relationship_spec` is in `existing_relationships`. If not, we try to create the
+    relationship represented by `relationship_spec`.
+    """
+    for existing_spec in existing_relationships:
+        relationships_match = relationship_spec.as_dict() == existing_spec.as_dict()
+        if relationships_match:
+            return None
```

Comment:
Nit: This `return None` feels odd, cos the function doesn't return an explicit value if it creates the relationship, and the caller never does anything with the return value. I'd just leave it at `return`.

Reply:
True! I'm used to mypy complaining about pure `return` statements (but that might have been in another scenario)....Ah the days of mypy.... _Shaking head to come back to reality_.